controllers/healthcheck_controller.go

```diff
@@ -86,6 +87,7 @@ type HealthCheckReconciler struct {
 	MaxParallel        int
 	RepeatTimersByName map[string]*time.Timer
 	workflowLabels     map[string]string
+	Locker             sync.RWMutex
```

Comment:
Minor nit: it might be nicer to either embed this lock if we want to use it generically in `HealthCheckReconciler` (so that you can simply call `r.Lock()` instead of `r.Locker.Lock()`, or possibly rename it if it's a lock specific for Timers e.g. `TimerLock`

Reply:
sure.

Reply:
yeah updating to TimerLock